app/src/main/java/com/greenfox/kalendaryo/MainActivity.java

```diff
@@ -1,143 +1,66 @@
 
-    @Override
-    public void onClick(View view) {
-        switch (view.getId()) {
-            case R.id.bn_logout:
-                signOut();
-                break;
-            case R.id.button2:
-                displayData();
-                break;
-            case R.id.choose_account:
-                Intent i = new Intent(this, LoginActivity.class);
-                i.putExtra("ifNewAccChoosen", true);
-                startActivity(i);
-                break;
-            case R.id.mergeCalsButton:
-                createMergedCals();
-        }
-    }
+            }
 
-    public void signOut() {
-        GoogleApiService.getInstance().getGoogleApiClient().connect();
-        GoogleApiService.getInstance().getGoogleApiClient().registerConnectionCallbacks(new GoogleApiClient.ConnectionCallbacks() {
             @Override
-            public void onConnected(@Nullable Bundle bundle) {
-                if(GoogleApiService.getInstance().getGoogleApiClient().isConnected()) {
-                    Auth.GoogleSignInApi.signOut(GoogleApiService.getInstance().getGoogleApiClient()).setResultCallback(new ResultCallback<Status>() {
-                        @Override
-                        public void onResult(@NonNull Status status) {
-                            if (status.isSuccess()) {
-                                Log.d("Log out", "User Logged out");
-                                GoogleApiService.finish();
-                                Intent intent = new Intent(MainActivity.this, LoginActivity.class);
-                                startActivity(intent);
-                                finish();
-                            }
-                        }
-                    });
-                }
+            public void onTabUnselected(TabLayout.Tab tab) {
+
             }
+
             @Override
-            public void onConnectionSuspended(int i) {
-                Log.d("Connection suspended", "Google API Client Connection Suspended");
+            public void onTabReselected(TabLayout.Tab tab) {
+
             }
         });
-        kalPref.clearAccountsAndAll();
-        startActivity(new Intent(MainActivity.this, LoginActivity.class));
-    }
-
-    private void checkSharedPreferencesForUser() {
-
-        ArrayList<String> accounts = kalPref.getAccounts();
-        if (accounts.size() == 0) {
-            Toast.makeText(this, "You have to log in", Toast.LENGTH_SHORT).show();
-            startActivity(new Intent(MainActivity.this, LoginActivity.class));
-        }
     }
 
-    public void displayData() {
-        ArrayList<String> accounts = kalPref.getAccounts();
-        for (int i = 0; i < accounts.size(); i++) {
-            KalAuth kalAuth = kalPref.getAuth(accounts.get(i));
-            myText.append(kalAuth.getEmail() + "\n");
-            token.append(kalAuth.getAccessToken() + "\n");
+    public void checkIfUserSignedIn () {
+        if (pref.clientToken().equals("")) {
```

Comment:
@BalazsSalfay `pref.isUserSignedIn()` and also this if can be directly in the oncreate